ui/helpers/utils/util.js

```diff
@@ -443,3 +443,77 @@ export const toHumanReadableTime = (t, milliseconds) => {
 export function clearClipboard() {
   window.navigator.clipboard.writeText('');
 }
+
+const solidityTypes = () => {
+  const types = [
+    'bool',
+    'address',
+    'string',
+    'bytes',
+    'int',
+    'uint',
+    'fixed',
+    'ufixed',
+  ];
+
+  const ints = Array.from(new Array(32)).map(
+    (_, index) => `int${(index + 1) * 8}`,
+  );
+  const uints = Array.from(new Array(32)).map(
+    (_, index) => `uint${(index + 1) * 8}`,
+  );
+  const bytes = Array.from(new Array(32)).map(
+    (_, index) => `bytes${index + 1}`,
+  );
+  return [...types, ...ints, ...uints, ...bytes];
+};
+
+export const sanitizeMessage = (msg, baseType, types) => {
+  let baseTypeDefinitions;
+  if (types) {
+    baseTypeDefinitions = types[baseType];
+    if (!baseTypeDefinitions) {
+      return msg;
```

Comment:
Is it possible for a signature request to get this far without any types, or with no primary type? I would hope this would be caught by our validation, but maybe not.

If this is possible, showing the user a message preview that could be totally wrong sounds dangerous :thinking:

Reply:
I agree, but not sure what to do in these instance.

Reply:
I have just confirmed that it is safe to assume here that `types` is defined, and that `primaryType` is included in `types`. So we should throw an error here for those cases.

This is guaranteed by the validation performed here: https://github.com/MetaMask/metamask-extension/blob/5e9ea9d8fad692d6b8346ccd72fc65a78e29164c/app/scripts/lib/typed-message-manager.js#L192

This is done before the message gets added to our state as an unapproved message. Validation is done [here](https://github.com/MetaMask/metamask-extension/blob/5e9ea9d8fad692d6b8346ccd72fc65a78e29164c/app/scripts/lib/typed-message-manager.js#L127), whereas the message isn't added until later ([here](https://github.com/MetaMask/metamask-extension/blob/5e9ea9d8fad692d6b8346ccd72fc65a78e29164c/app/scripts/lib/typed-message-manager.js#L143)).